blocks/monitoring/evaluators.py

```diff
@@ -211,8 +248,23 @@ def process_batch(self, batch):
         if self._accumulate_fun is not None:
             self._accumulate_fun(**batch)
 
+        try:
+            batch = dict_subset(batch, self.buffer2_.input_names)
+        except KeyError:
+            reraise_as(
+                "Not all data sources required for monitoring were"
+                " provided. The list of required data sources:"
+                " {}.".format(self.buffer2_.input_names))
+        if self._accumulate_fun2 is not None:
+            self.non_theano_variables[-1].accumulate(self._accumulate_fun2(**batch)[0], self._accumulate_fun2(**batch)[1])
```

Comment:
There are many ways of doing it, just chose whichever one is easiest for you. A useful method to keep in mind is `list.index(item)`, which would allow you to do something like 
```python
monitored_quantity.accumulate(*[numerical_values[unique_theano_variables.index(requirement)]
                                for requirement in monitored_quantity.requires])
```
and avoid using a lookup dictionary entirely.